packages/salesforcedx-vscode-core/src/commands/stopApexDebugLogging.ts

```diff
@@ -1,90 +1,63 @@
-
-    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
-    execution.processExitSubject.subscribe(async data => {
-      this.logMetric(execution.command.logName, startTime);
-      if (data !== undefined && String(data) === '0') {
-        developerLogTraceFlag.turnOffLogging();
-        hideTraceFlagExpiration();
-      }
-    });
-  }
-}
+const command = 'SFDX: Turn Off Apex Debug Log for Replay Debugger';
 
 export const turnOffLogging = async (): Promise<void> => {
+  console.log('Enter turnOffLogging()');
+  const channelService = new ChannelService(OUTPUT_CHANNEL);
+  handleStartCommand(channelService, command);
+
   if (developerLogTraceFlag.isActive()) {
-    const execution = new CliCommandExecutor(deleteTraceFlag(), {
-      cwd: workspaceUtils.getRootWorkspacePath()
-    }).execute();
-    telemetryService.sendCommandEvent(execution.command.logName);
-    const resultPromise = new CommandOutput().getCmdResult(execution);
-    const result = await resultPromise;
-    const resultJson = JSON.parse(result);
-    if (resultJson.status === 0) {
-      return Promise.resolve();
-    } else {
-      return Promise.reject('Restoring the debug levels failed.');
-    }
-  }
-};
+    console.log('Developer log trace flag is active');
+    try {
+      await vscode.window.withProgress(
+        {
+          location: vscode.ProgressLocation.Notification,
+          title: command,
+          cancellable: false
+        },
+        async progress => {
+          progress.report({ message: 'Running...' });
+          const nonNullTraceFlag = developerLogTraceFlag.getTraceFlagId()!;
+          const connection = await WorkspaceContext.getInstance().getConnection();
+          await TraceFlagsRemover.getInstance(connection).removeTraceFlag(nonNullTraceFlag);
+          telemetryService.sendCommandEvent('stop_apex_debug_logging');
+        }
+      );
 
-const deleteTraceFlag = (): Command => {
-  const nonNullTraceFlag = developerLogTraceFlag.getTraceFlagId()!;
-  return new SfCommandBuilder()
-    .withDescription(nls.localize('stop_apex_debug_logging'))
-    .withArg('data:delete:record')
-    .withFlag('--sobject', 'TraceFlag')
-    .withFlag('--record-id', nonNullTraceFlag)
-    .withArg('--use-tooling-api')
-    .withLogName('stop_apex_debug_logging')
-    .build();
-};
-class ActiveLogging implements ParametersGatherer<{}> {
-  public async gather(): Promise<CancelResponse | ContinueResponse<{}>> {
-    if (developerLogTraceFlag.isActive()) {
-      return { type: 'CONTINUE', data: {} };
+      await handleFinishCommand(
+        channelService,
+        command,
+        true
+      );
+    } catch (error) {
+      console.error('Error in turnOffLogging(): ', error);
+      telemetryService.sendException('stop_apex_debug_logging', error);
+      await handleFinishCommand(
+        channelService,
+        command,
+        false
+      );
+      return Promise.reject('Restoring the debug levels failed.');
```

Comment:
instead of returning a reject, what happens if we throw a new Error?  As is, it's the equivalent of throwing a string instead of an error.